tailord/src/dbus/performance.rs

```diff
@@ -0,0 +1,51 @@
+use zbus::{dbus_interface, fdo};
+
+use crate::performance::PerformanceProfileRuntimeHandle;
+
+pub struct PerformanceInterface {
+    pub handler: Option<PerformanceProfileRuntimeHandle>,
+}
+
+impl PerformanceInterface {
+    fn handler(&self) -> fdo::Result<&PerformanceProfileRuntimeHandle> {
+        self.handler.as_ref().ok_or(fdo::Error::Failed(
+            "No performance profile handler available".to_string(),
+        ))
+    }
+
+    fn handler_mut(&mut self) -> fdo::Result<&mut PerformanceProfileRuntimeHandle> {
+        self.handler.as_mut().ok_or(fdo::Error::Failed(
+            "No performance profile handler available".to_string(),
+        ))
+    }
+}
+
+#[dbus_interface(name = "com.tux.Tailor.Performance")]
+impl PerformanceInterface {
+    async fn set_profile(&mut self, name: &str) -> fdo::Result<()> {
+        self.handler()?
+            .profile_sender
+            .send(name.to_string())
+            .await
+            .map_err(|err| {
+                fdo::Error::IOError(format!("unable to set performance profile {name}: {err}"))
+            })?;
+        self.handler_mut()?.set_active_performance_profile(name);
+        Ok(())
+    }
+
+    async fn get_profile(&self) -> fdo::Result<String> {
+        Ok(self.handler()?.get_active_performance_profile().to_string())
+    }
+
```

Comment:
I wonder where the getters and setters here would be useful. Neither the LED nor the fan interfaces have such a thing IIRC (they only have methods for short overwrites, which are used by the GUI to give the user feedback). 
Usually, updating a profile and reloading it would be the proper way of setting the performance profile because it makes changes persist across system restart, whereas those methods are not touching the configuration.

Reply:
I added getters and setter because they are nice to use in the cli DBUS interface. They let you change things without changing the overall profile, its a sort of override.

Reply:
Let me document a little what I did, that is probably a good idea.